src/components/CommunityAnnotationTable/components/CuratorTableCell.tsx

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import * as R from 'ramda';
+import { Icon, Popup } from 'semantic-ui-react';
+
+import { makeTableColumnData } from '../model';
+
+type CuratorTableCellProps = {
+    value: {
+        validated: boolean;
+        curator_id: string;
+        curator_name: string;
+    };
+};
+
+class CuratorTableCell extends React.Component<CuratorTableCellProps> {
+    constructor(props: CuratorTableCellProps) {
+        super(props);
+    }
+
+    render() {
+        const {
+            value: { validated, curator_id, curator_name },
+        } = this.props;
+
+        const orcidIDRegex = /(?:\d{4}-){3}\d{3}[0-9,X]/;
+
+        let iconName,
+            iconColor,
+            popupText = '';
+
+        if (validated && orcidIDRegex.test(curator_id)) {
+            iconName = 'check circle outline';
+            iconColor = 'green';
+            popupText = 'This annotation was generated on this site.';
+        } else if (orcidIDRegex.test(curator_id)) {
+            iconName = 'times circle outline';
+            iconColor = 'red';
+            popupText = 'This annotation was NOT generated on this site.';
+        } else {
+            iconName = 'laptop';
+            iconColor = 'orange';
+            popupText =
+                'This annotation is not linked to an ORCID iD and is therefore likely a prediction from a tool.';
+        }
```

Comment:
Can this be a separate function?

Reply:
Not too sure how put this nicely into a function. Do you have a suggestion ?

Reply:
Eh. It's fine